src/Components/Components/src/RuntimeHelpers.cs

```diff
@@ -17,6 +18,43 @@ public static class RuntimeHelpers
         /// <typeparam name="T"></typeparam>
         /// <param name="value"></param>
         /// <returns></returns>
+        //
+        // This method is used to create an expression binding site with a compile-time known type. This helps with providing
+        // good error messages, as well as proper method-group-to-delegate conversion when assigning component parameters.
         public static T TypeCheck<T>(T value) => value;
+
+        /// <summary>
+        /// Not intended for use by application code.
+        /// </summary>
+        /// <param name="receiver"></param>
+        /// <param name="callback"></param>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        //
+        // This method is used with `@bind-Value` for components. When a component has a generic type, it's
+        // really messy to write to try and write the parameter type for ValueChanged - because it can contain generic
+        // type parameters. We're using a trick of type inference to generate the proper typing for the delegate
+        // so that method-group-to-delegate conversion works.
+        public static EventCallback<T> CreateInferredEventCallback<T>(object receiver, Action<T> callback, T value)
```

Comment:
Need to add a new method first, then we can remove the old one after the compiler updates.